### PR TITLE
LibJS: Refactor CalendarFields for better linearity

### PR DESCRIPTION
**Pull request by moustafaraafat**
Ticks off one box in https://github.com/SerenityOS/serenity/issues/15525

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 1]
```diff
@@ -106,17 +106,19 @@ ThrowCompletionOr<Vector<String>> calendar_fields(VM& vm, Object& calendar, Vect
     // 1. Let fields be ? GetMethod(calendar, "fields").
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:106]: You seem to be referencing the wrong commit
I believe it should be this one: https://github.com/tc39/proposal-temporal/commit/9b139a1d2d42c04f5b755c73cdbb80b9333b298b

Which is the one referenced in #15525

**Review by davidot — CHANGES_REQUESTED** [review]
Code looks fine, just a comment about the commit message referencing the wrong commit

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 2]
```diff
@@ -106,17 +106,19 @@ ThrowCompletionOr<Vector<String>> calendar_fields(VM& vm, Object& calendar, Vect
...
     for (auto& field_name : field_names)
         field_names_values.append(js_string(vm, field_name));
-    Value fields_array = Array::create_from(realm, field_names_values);
-
-    // 3. If fields is not undefined, then
-    if (fields) {
-        // a. Set fieldsArray to ? Call(fields, calendar, « fieldsArray »).
-        fields_array = TRY(call(vm, *fields, &calendar, fields_array));
-    }
+    Value fields_array = TRY(call(vm, *fields, &calendar, Array::create_from(realm, field_names_values)));
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:121]: `auto`, also you can replace the manual loop with

```cpp
Array::create_from<StringView>(realm, field_names, [&](auto value) { return js_string(vm, value); })
```

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:106]: Yeah, please always reference commits from the main branch, not the PR :)

Comment (moustafaraafat) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:106]: Sure, sorry my bad.

**Review by davidot — APPROVED** [review]
Thanks for fixing the commit!
LGTM

**Pull request merged**